### PR TITLE
Changes to skip running DDL. Mainly used when running fault tolerance…

### PR DESCRIPTION
**Pull request by aravind-nallan-yb**
… tests with nodes down.

### CODE REVIEW

File: src/main/java/com/yugabyte/sample/apps/AppBase.java [hunk 1]
```diff
@@ -209,14 +209,16 @@ protected Connection getPostgresConnection(String database, String username, Str
...
   protected static void createKeyspace(Session session, String ks) {
-    String create_keyspace_stmt = "CREATE KEYSPACE IF NOT EXISTS " + ks +
-      " WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor' : 1};";
-    // Use consistency level ONE to allow cross DC requests.
-    session.execute(
-      session.prepare(create_keyspace_stmt)
-        .setConsistencyLevel(ConsistencyLevel.ONE)
-        .bind());
-    LOG.debug("Created a keyspace " + ks + " using query: [" + create_keyspace_stmt + "]");
+    if (!appConfig.skipDDL) {
```
Comment (tedyu) [inline src/main/java/com/yugabyte/sample/apps/AppBase.java:212]: DDL covers not just keyspace creation.

What about createTablesIfNeeded() ?

If node is down, wouldn't DML be affected as well ?

Comment (ncarvind) [reply src/main/java/com/yugabyte/sample/apps/AppBase.java:212]: See changes above to Main.java

File: src/main/java/com/yugabyte/sample/apps/AppBase.java [hunk 2]
```diff
@@ -572,6 +574,7 @@ public void dropCassandraTable(String tableName) {
    * @throws java.lang.Exception in case of CREATE statement errors.
    */
   public void createTablesIfNeeded(TableOp tableOp) throws Exception {
+    LOG.info("Creating Cassandra tables...");
```
Comment (nchandrappa) [inline src/main/java/com/yugabyte/sample/apps/AppBase.java:577]: Can we please change from `Cassandra` to `YCQL` when logging.

**Review by nchandrappa — COMMENTED** [review]
Hello @aravind-nallan-yb, Code changes look good to me. Have one small review comment, please take a look.

Before we merge the changes, just wanted to understand the need for this flag. From the PR title, I couldn't get any context on the change being made. If you see the DDL statements which the app uses has `IF NOT EXISTS`, so how are we planning to use the `skipDDL` flag. 

Thanks.

File: src/main/java/com/yugabyte/sample/apps/AppBase.java (hunk 2 above)
Comment (aravind-nallan-yb) [reply src/main/java/com/yugabyte/sample/apps/AppBase.java:577]: Sure

**Comment by aravind-nallan-yb** [comment]
> Hello @aravind-nallan-yb, Code changes look good to me. Have one small review comment, please take a look.
> 
> Before we merge the changes, just wanted to understand the need for this flag. From the PR title, I couldn't get any context on the change being made. If you see the DDL statements which the app uses has `IF NOT EXISTS`, so how are we planning to use the `skipDDL` flag.
> 
> Thanks.

Hey Nikhil,

Please see https://docs.google.com/document/d/1brgDJC_v8QIguPo_FLxM0C1WkNFqtXG-tlUB9-NZNiA/edit?usp=sharing for full context.

But TL;DR is when i was trying fault tolerance scenarios where some nodes are down even if you include IF NOT EXISTS, any DDL just hangs.

**Comment by nchandrappa** [comment]
Got it, Thanks for the context. Now I understand the use case for `skipDDL`. 

I'll approve the PR. Thanks

**Review by nchandrappa — COMMENTED** [review]
LGTM.

**Pull request merged**